Sources/AppcuesKit/Data/Models/Experience.swift

```diff
@@ -51,6 +51,44 @@ internal struct FailedExperience: Decodable {
     }
 }
 
+// a helper that deserializes a collection of traits and also enforces that
+// no trait types are duplicated within the collection
+internal struct TraitCollection: Decodable {
+    var traits: [Experience.Trait]
+
+    init(from decoder: Decoder) throws {
+        var container = try decoder.unkeyedContainer()
+
+        var traits: [Experience.Trait] = []
+        var traitTypes = Set<String>()
+        var duplicates = Set<String>()
+
+        if let count = container.count {
+            traits.reserveCapacity(count)
+        }
+
+        while !container.isAtEnd {
+            let trait = try container.decode(Experience.Trait.self)
+            if traitTypes.insert(trait.type).inserted {
+                // normal case, capture the decoded trait for our resulting array
+                traits.append(trait)
+            } else {
+                // a dupe was found and this will end up causing a decoding error below
+                duplicates.insert(trait.type)
```

Comment:
I like that we're collecting all duplicates rather than failing on the first one 👍